improve: !map webhooks 🌊

mistralwz opened the pull request:
the current one is very not pretty (it's bad because on the last commit I could not test it at all...)

I've made the embed a lot more compact but still easy to understand - clicking on the map name lets you download the map, linking more stuff, displaying extra stats. might be a bit bloated... but nicer than the current one.

might change it up a bit more after bns give more inputs

<details>
<summary>screenshot</summary>

![](https://cdn.discordapp.com/attachments/1165545768917139536/1170356306138374204/image.png)

</details>

mistralwz committed:
improve: !map webhooks 🌊

the current one is very not pretty (it's bad because on the last commit I could not test it at all...)

I've made the embed a lot more compact but still easy to understand - clicking on the map name lets you download the map, linking more stuff, displaying extra stats. might be a bit bloated...

![''](https://cdn.discordapp.com/attachments/1164100886235185162/1170122757367922749/Screenshot_2023-11-04_050428.png)
![''](https://cdn.discordapp.com/attachments/1164100886235185162/1170122757367922749/Screenshot_2023-11-04_050428.png)
![''](https://cdn.discordapp.com/attachments/1164100886235185162/1170122757770588200/Screenshot_2023-11-04_050337.png)

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mistralwz committed:
more changes to the embed

mistralwz committed:
removed map stats icons

james was right it doesn't look that good

mistralwz committed:
oops (forgot to replace map with {message[1]}

tsunyoku commented on constants/chatbotCommands.py:
```diff
@@ -1440,30 +1440,53 @@ async def editMap(fro: str, chan: str, message: list[str]) -> Optional[str]:
+        author_url=f"{nominator_profile_url}",
+        author_icon=f"https://a.akatsuki.gg/{token['user_id']}",
+        title=f'{mode_to_emoji(res["mode"])} {res["song_name"]}',
+        title_url=f'https://bathbot.de/osudirect/{res["beatmapset_id"]}',
```
idk about bathbot as the download... is your intention to download an osz or to redirect to ingame direct? cus we can do both without needing bathbot

mistralwz replied:
we're redirecting it to ingame direct! afaik you can't use `osu://` links on the embed - if that was an option i think bathbot wouldn't have made that site.

tsunyoku replied:
arghhhhhhhhhhh i hate more dependencies on random websites i kinda would rather it just not go to direct

mistralwz replied:
that's ok! let's just make that redirect to the official bancho/akatsuki listing then? i think that seems to make the most sense

tsunyoku replied:
yes

mistralwz replied:
changed things in https://github.com/osuAkatsuki/bancho-service/pull/92/commits/459d1a09985d1e6500fcb075815eef46cf7313d6 to suggested changes

mistralwz replied:
@tsunyoku im going to sleep but i think everything looks good now, here's the current embed...

![](https://cdn.discordapp.com/attachments/1165545768917139536/1171197086709055578/image.png)

mistralwz committed:
update beatconnect emoji

now uses the beatconnect emoji in their server and not the one in akatsuki's discord

mistralwz committed:
Merge branch 'main' into main

mistralwz committed:
suggestion fixes

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mistralwz committed:
Merge branch 'main' into main

mistralwz committed:
update osu.direct emoji

mistralwz committed:
Merge branch 'main' of https://github.com/mistralwz/bancho-service
